### PR TITLE
Feat/ingredients controller

### PR DESCRIPTION
**Pull request by isadorabk**

### CODE REVIEW

File: controllers/ingredients.controller.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+const db = require('../models').db;
+const filterProps = require('../services/utils.js').filterProps;
+
+
+class IngredientsController {
+  constructor (ingredientModel) {
+    if (!ingredientModel) throw new Error('Ingredient model not provided');
+    this.Ingredient = ingredientModel;
+    this.createIngredient = this.createIngredient.bind(this);
```
Comment (arol) [inline controllers/ingredients.controller.js:11]: Just FYI: if you use BABEL, you can install the plugin [transform class properties](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/) to do something like

```js
createIngredient = async () => {
   …
}
```

So you won't need to bind it. It's just another syntax, nothing wrong with your approach.

File: controllers/ingredients.controller.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    if (ingredients) {
+      ctx.body = [];
+      ingredients.forEach(el => {
+        const ingredient = {
+          id: el.id,
+          name: el.name,
+          ingredient_type: el.dataValues.Ingredient_type.dataValues.name
+        };
+        ctx.body.push(ingredient);
+      });
```
Comment (arol) [inline controllers/ingredients.controller.js]: These lines 64-72 can be 

```js
ctx.body = ingredients.map(el => ({
  id: el.id,
  name: el.name,
  ingredient_type: el.dataValues.Ingredient_type.dataValues.name
})
```

File: controllers/ingredients.controller.js [hunk 3]
```diff
@@ -61,14 +61,12 @@ class IngredientsController {
     });
 
     if (ingredients) {
-      ctx.body = [];
-      ingredients.forEach(el => {
-        const ingredient = {
+      ctx.body = ingredients.map(el => {
```
Comment (arol) [inline controllers/ingredients.controller.js]: Nice refactor 👍

**Review by arol — DISMISSED** [review]

**Review by nickschoey — APPROVED** [review]
I'm approving this review because Isadora is threatening me and against my will.

**Pull request merged**